includes/class-omise-money.php

```diff
@@ -0,0 +1,56 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * @since 3.5
+ */
+class Omise_Money {
+	/**
+	 * @var array
+	 */
+	private static $subunit_multiplier = array(
+		'AUD' => 100,
+		'CAD' => 100,
+		'CHF' => 100,
+		'CNY' => 100,
+		'DKK' => 100,
+		'EUR' => 100,
+		'GBP' => 100,
+		'HKD' => 100,
+		'JPY' => 1,
+		'MYR' => 100,
+		'SGD' => 100,
+		'THB' => 100,
+		'USD' => 100
+	);
+
+	/**
+	 * @param  int|float|string $amount
+	 * @param  string           $currency
+	 *
+	 * @return int
+	 */
+	public function to_subunit( $amount, $currency ) {
+		$amount   = static::purify_amount( $amount );
+		$currency = strtoupper( $currency );
+
+		if ( ! isset( static::$subunit_multiplier[ $currency ] ) ) {
```

Comment:
If we're insisting on keeping the currency check (which I still think really shouldn't be here) then we should really split it out into a `currency_supported` function? As it will also be used in `subunit_to_amount`

Reply:
I would like to consider having that after we really implement that `subunit_to_amount` method (It's a good practice to reduce the duplicated code, but not yet to happened at the moment).